runtime/startup_byt.c

```diff
@@ -554,9 +562,13 @@ CAMLexport void caml_main(char_os **argv)
   /* Build the table of primitives */
   shared_lib_path = read_section_to_os(fd, &trail, "DLPT");
   shared_libs = read_section_to_os(fd, &trail, "DLLS");
-  req_prims = read_section(fd, &trail, "PRIM");
+  req_prims = read_section(fd, &trail, "PRIM", NULL);
+  symb_section = read_section(fd, &trail, "SYMB", &symb_section_len);
+  crcs_section = read_section(fd, &trail, "CRCS", &crcs_section_len);
   if (req_prims == NULL) caml_fatal_error("no PRIM section");
```

Comment:
The check for `req_prims == NULL` should stay next to the assignment to `req_prims`. (diff doesn't do a good job on this one)
```suggestion
  req_prims = read_section(fd, &trail, "PRIM", NULL);
  if (req_prims == NULL) caml_fatal_error("no PRIM section");
  symb_section = read_section(fd, &trail, "SYMB", &symb_section_len);
  crcs_section = read_section(fd, &trail, "CRCS", &crcs_section_len);
```